proposed/http-middleware/middleware-meta.md

```diff
@@ -270,75 +276,37 @@ additional information._
 
 [promises]: https://promisesaplus.com/
 
-### 5.2 Delegate Design
-
-The `DelegateInterface` defines a single method that accepts a request and
-returns a response. The delegate interface must be implemented by any middleware
-dispatcher that uses middleware implementing `MiddlewareInterface`.
-
-#### Why the term "delegate"?
-
-The term "delegate" means something designated to act for or represent another.
-In terms of middleware design, a delegate is called upon by middleware when the
-middleware is unable to process the request itself; the delegate then processes
-the request _for the original middleware_ in order to return a response.
-
-#### Why isn't the delegate a `callable`?
-
-Using an interface type hint improves runtime safety and IDE support.
-
-_See "discussion of FrameInterface" in [relevant links](#8-relevant-links) for
-additional information._
-
-#### Why doesn't delegate use `__invoke`?
+### 5.2 Handler Design
 
-Classes that define `__invoke` can be more loosely type hinted as
-`callable`, which results in less strict typing. This is generally undesirable,
-especially when the method uses strict typing.
+The `HandlerInterface` defines a single method that accepts a request and
+MUST return a response. The handler MAY delegate to another handler.
 
-#### Why does the delegate conflict with middleware?
-
-Both the middleware and delegate interface define a `process` method to
-discourage misuse of middleware as delegates.
-
-The implementation of the delegate should be defined within middleware
-dispatching systems.
-
-#### Why not the term `$next`?
-
-Several existing middleware libraries use the term `$next` instead of
-`$delegate`. `$next` implies an action: "Next, please!" As such, these libraries
-define `$next` as a `callable`, which we note was undesirable for purposes of
-this specification in the previous section.
-
-Additionally, since we are defining an object, we chose to use a noun instead of
-a verb to name the interface.
+#### Why is a server request required?
 
-Further, we did not choose the term `next` for the action delegates invoke, as
-that verb implies a queue or stack. The delegate is not required to implement
-either pattern internally in order to do its work; its only job is to _process_
-the request to return a response.
+To make it clear that the handler can only be used in a synchronous, server
```

Comment:
Is the word "synchronous" needed here? If so (I think it is), we need to explain _why_, and how it might differ from an async architecture (where a promise would likely be returned instead of a response).